src/model_repository_manager/model_lifecycle.cc

```diff
@@ -513,7 +513,15 @@ ModelLifeCycle::AsyncLoad(
     // Load model asynchronously via thread pool
     load_pool_->Enqueue([this, model_id, version, model_info, OnComplete,
                          load_tracker, is_config_provided]() {
-      CreateModel(model_id, version, model_info, is_config_provided);
+      for (size_t retry = 0; retry <= options_.load_retry; ++retry) {
+        model_info->state_ = ModelReadyState::LOADING;
+        CreateModel(model_id, version, model_info, is_config_provided);
+        // Model state will be changed to NOT loading if failed to load,
+        // so the model is loaded if state is LOADING.
+        if (model_info->state_ == ModelReadyState::LOADING) {
```

Comment:
question - is there a pause / is there one needed between retry attempts?

Reply:
Why would one be needed?

Reply:
typically you after some time in order to let a possible transient error resolve - instead of retrying immediately